openedx/core/djangoapps/verified_track_content/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,268 @@
+                course_id=rerun_course_key,
+                mode_slug=CourseMode.VERIFIED
+            )
+        except CourseMode.DoesNotExist:
+            raise CommandError("Verified CourseMode is not defined for course: '%s'" % rerun_course_key)
+
+        items = module_store.get_items(rerun_course_key)
+        items_to_update = []
+
+        all_cohorted_track_group_ids = set()
+        for audit_course_user_group_partition_group in random_audit_course_user_group_partition_groups:
+            all_cohorted_track_group_ids.add(audit_course_user_group_partition_group.group_id)
+            all_cohorted_track_group_ids.add(verified_course_user_group_partition_group.group_id)
+        if not items:
+            raise CommandError("Items for Course with key '%s' not found." % rerun_course_key)
+
+        for item in items:
+            # Verify that there exists group access for this xblock, otherwise skip these checks
+            if item.group_access:
+                set_audit_enrollment_track = False
+                set_verified_enrollment_track = False
+
+                # Check the partition and group IDs for the audit course groups, if they exist in
+                # the xblock's access settings then set the audit track flag to true
+                for audit_course_user_group_partition_group in random_audit_course_user_group_partition_groups:
+                    audit_partition_group_access = item.group_access.get(
+                        audit_course_user_group_partition_group.partition_id,
+                        None
+                    )
+                    if (audit_partition_group_access
+                            and audit_course_user_group_partition_group.group_id in audit_partition_group_access):
+                        set_audit_enrollment_track = True
+
+                # Check the partition and group IDs for the verified course group, if it exists in
+                # the xblock's access settings then set the verified track flag to true
+                verified_partition_group_access = item.group_access.get(
+                    verified_course_user_group_partition_group.partition_id,
+                    None
+                )
+                if (verified_partition_group_access
+                        and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                    # If the item has group_access that is not the
+                    # verified or audit group IDs then raise an error
+                    # This only needs to be checked for this partition_group once
+                    non_verified_track_access_groups = set(verified_partition_group_access) \
+                                                       - all_cohorted_track_group_ids
+                    if non_verified_track_access_groups:
+                        errors.append(
+                            "Non audit/verified cohorted content group set for xblock, location '%s' with IDs '%s'"
```

Comment:
This logic doesn't seem to be sufficient. I reran this course, and the component that was linked to both the Audit Content Group and another content group didn't trigger the error condition. It ended up just assigning the component to the Audit enrollment track.

https://studio-jlajoie.sandbox.edx.org/container/block-v1:christina+1+1+type@vertical+block@e22aaa3d37d3404f917a6070841c9166